src/_modules/experimental/Request.sol

```diff
@@ -371,6 +446,83 @@ library LibResponse {
     // }
 }
 
+library LibHeaders {
+    function toJsonObject(RequestHeaders self) internal pure returns (JsonObject memory obj) {
+        assembly {
+            obj := self
+        }
+    }
+
+    function toRequestHeaders(JsonObject memory obj) internal pure returns (RequestHeaders headers) {
+        assembly {
+            headers := obj
+        }
+    }
+
+    function set(RequestHeaders self, string memory key, string memory value) internal returns (RequestHeaders) {
+        string[] memory values = new string[](1);
+        values[0] = value;
+
+        return set(self, key, values);
+    }
+
+    function set(RequestHeaders self, string memory key, string[] memory values) internal returns (RequestHeaders) {
```

Comment:
should we have set/insert + append instead? (if not lets justify it here for reference)

Reply:
yes! That's something I wanted to discuss. I think `set` should replace the header instead of appending them

Reply:
set == insert, it is just the naming that reqwest uses. I was suggesting we might want to rename it considering everything else is pretty much the same as reqwest's api

Reply:
https://docs.rs/reqwest/latest/reqwest/header/struct.HeaderMap.html#method.insert